be/src/exprs/string_functions.cpp

```diff
@@ -2075,6 +2079,58 @@ StatusOr<ColumnPtr> StringFunctions::unhex(FunctionContext* context, const starr
     return VectorizedStringStrictUnaryFunction<unhexImpl>::evaluate<TYPE_VARCHAR, TYPE_VARCHAR>(columns[0]);
 }
 
+DEFINE_STRING_UNARY_FN_WITH_IMPL(url_encodeImpl, str) {
+    return StringFunctions::url_encode_func(str.to_string());
+}
+
+std::string StringFunctions::url_encode_func(const std::string& value) {
+    std::ostringstream escaped;
+    escaped.fill('0');
+    escaped << std::hex;
+
+    for (std::string::const_iterator i = value.begin(), n = value.end(); i != n; ++i) {
+        std::string::value_type c = (*i);
+        if (isalnum(c) || c == '-' || c == '_' || c == '.' || c == '~') {
+            escaped << c;
+            continue;
+        }
+
+        escaped << std::uppercase;
+        escaped << '%' << std::setw(2) << int((unsigned char)c);
+        escaped << std::nouppercase;
+    }
+    return escaped.str();
+}
+
+StatusOr<ColumnPtr> StringFunctions::url_encode(FunctionContext* context, const starrocks::Columns& columns) {
+    return VectorizedStringStrictUnaryFunction<url_encodeImpl>::evaluate<TYPE_VARCHAR, TYPE_VARCHAR>(columns[0]);
+}
+
+DEFINE_STRING_UNARY_FN_WITH_IMPL(url_decodeImpl, str) {
+    return StringFunctions::url_decode_func(str.to_string());
+}
+
+std::string StringFunctions::url_decode_func(const std::string& value) {
+    std::string ret;
+    char ch;
+    int i, ii;
+    for (i = 0; i < value.length(); i++) {
+        if (int(value[i]) == 37) {
+            sscanf(value.substr(i + 1, 2).c_str(), "%x", &ii);
```

Comment:
what about use `      std::istringstream is(in.substr(i + 1, 2))` ?

Reply:
What if `value.substr(i + 1, 2).c_str()` is illegal hex? return null or throw exception?